engine/core/src/test/java/org/pentaho/reporting/engine/classic/core/layout/text/TextUtilityTest.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2001 - 2018 Object Refinery Ltd, Hitachi Vantara and Contributors..  All rights reserved.
+ */
+
+package org.pentaho.reporting.engine.classic.core.layout.text;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.pentaho.reporting.engine.classic.core.layout.model.RenderableText;
+import org.pentaho.reporting.libraries.fonts.registry.BaselineInfo;
+import org.pentaho.reporting.libraries.fonts.registry.FontMetrics;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Mockito.*;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.junit.runner.RunWith;
+
+
+@RunWith( PowerMockRunner.class )
+@PrepareForTest( TextUtility.class )
+public class TextUtilityTest {
+
+  @Test
```

Comment:
.. because the TextUtilityMethod is a pure function and only depends on the inputs that you pass in. 

The function you want to test is the 'createPaddedBaselineInfo' which takes two parameters that are complex. Ignore the 'BaselineInfo' parameter, as this is a final class and just a data carrier. Just instantiate one and set some sensible values there.

The only thing you need to actually mock is the font metrics object. "FontMetrics#getBaselines" can simply return your BaselineInfo object from above, and the other three methods used can be easily stubbed, as you do in the code here. 

The static method that you annotated with that Guava annotation does not need to annotated at all. Its behaviour is controlled by the dominant-baseline method returned via the BaselineInfo object. Just set that property to 'MIDDLE' in your BaselineInfo object (that you return as pass-through value from the font metrics mock) when you set up the object and it will return the right value for you. In any case, that dominant baseline is only passed on to the resulting ExtendedBaselineInfo and has no bearing on the logical flow of your method under test. So set up your tests that it returns something sensible, but the exact details of that value wont matter.